Demo/src/main/java/com/paypal/android/ui/paypal/PayPalNativeFragment.kt

```diff
@@ -24,45 +24,44 @@ class PayPalNativeFragment : Fragment() {
     @Inject
     lateinit var sdkSampleServerApi: SDKSampleServerApi
 
-    private val viewModel: PayPalNativeViewModel by viewModels()
+    private var selectedShippingPreference: ShippingPreferenceType? = null
 
-    private var selectedOptionId = R.id.order_checkout
+    private val viewModel: PayPalNativeViewModel by viewModels()
 
     override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View {
-
         binding = FragmentPayPalNativeBinding.inflate(inflater, container, false)
         viewModel.state.observe(viewLifecycleOwner) { viewState ->
             checkViewState(viewState)
         }
         with(binding) {
+            startNativeCheckout.setOnClickListener { startCheckout() }
             fetchAccessTokenButton.setOnClickListener { viewModel.fetchAccessToken() }
-            startNativeCheckout.setOnClickListener { onStartNativeCheckoutClicked() }
-            orderCheckout.setOnClickListener { onRadioButtonClicked(it) }
-            orderIdCheckout.setOnClickListener { onRadioButtonClicked(it) }
-            billingAgreementCheckout.setOnClickListener { onRadioButtonClicked(it) }
-            vaultCheckout.setOnClickListener { onRadioButtonClicked(it) }
             tryAgainButton.setOnClickListener { viewModel.reset() }
         }
+        initShippingOptions()
         return binding.root
     }
 
-    private fun onRadioButtonClicked(view: View) {
-        if (view is MaterialRadioButton && view.isChecked) {
-            selectedOptionId = view.id
+    private fun startCheckout() {
+        selectedShippingPreference?.let {
+            viewModel.orderIdCheckout(it)
         }
-    }
-
-    private fun onStartNativeCheckoutClicked() {
-        when (selectedOptionId) {
-            R.id.order_checkout -> { viewModel.orderCheckout() }
-            R.id.order_id_checkout -> { viewModel.orderIdCheckout() }
-            R.id.billing_agreement_checkout -> { viewModel.billingAgreementCheckout() }
-            R.id.vault_checkout -> { viewModel.vaultCheckout() }
-            else -> {}
+        binding.checkoutOptionsRadioGroup.isVisible = false
+    }
+
+    private fun initShippingOptions() {
+        ShippingPreferenceType.values().forEach { shippingPreferenceType ->
+            val radioButton = android.widget.RadioButton(requireContext())
```

Comment:
```suggestion
            val radioButton = RadioButton(requireContext())
```
nit: noticed a fully qualified name here.